box-share-sdk/src/main/java/com/box/androidsdk/share/vm/DataWrapper.java

```diff
@@ -0,0 +1,55 @@
+package com.box.androidsdk.share.vm;
+
+import androidx.annotation.StringRes;
+
+/**
+ * A data wrapper to make sure LiveData changes can take network responses into account
+ * @param <T> the data type of the item in the response that will be returned
+ */
+public class DataWrapper<T> {
+    protected T mData;
+    protected int mStrRes;
+    public static final int SUCCESS = -1;
+
+    /**
+     * Updates the item with the data passed in. Use this if the request was successful.
+     * @param data the data from the result
+     */
+    public void success(T data) {
```

Comment:
I'd call this setData

Reply:
Following a modified version of this pattern here. I also added another method for success that also takes in a string resource just in case we want to display message for success case in the future.

https://github.com/googlesamples/android-architecture-components/blob/master/GithubBrowserSample/app/src/main/java/com/android/example/github/vo/Resource.kt